ShopAppTests/Data/API/Alamofire/BaseAPISpec.swift

```diff
@@ -82,10 +84,13 @@ class BaseAPISpec: QuickSpec {
                     
                     waitUntil { done in
                         api.execute(request) { (response, error) in
-                            expect((response as? [String: Any])?[key] as? String) == value
-                            expect(error).to(beNil())
+                            self.checkResponse(response, error: error, key: key, value: value)
                             
-                            done()
+                            api.execute(request) { (response, error) in
```

Comment:
What is the purpose of sending 2 requests and what do you test with them?

Reply:
We cover first callback in response method of Base API class (line 36-40) by the second call of execute method in this test.

Reply:
I don't understand how it helps to test cache. For example, if I remove lines 34-40 of BaseAPI class - core functionality of caching API - this test is still passing.